transformer_engine/common/comm_gemm_overlap/comm_gemm_overlap.cpp

```diff
@@ -591,6 +597,29 @@ void CommOverlapBase::split_overlap_rs(const TensorWrapper &A, bool transa, cons
   NVTE_CHECK_CUDA(cudaStreamWaitEvent(stream_main, _stop_comm, 0));
 }  // CommOverlapBase::split_overlap_rs
 
+void CommOverlapBase::bulk_overlap_external_ag(const TensorWrapper &input, cudaStream_t send_stream,
+                                               cudaStream_t recv_stream, cudaStream_t stream_main) {
+  int comm_bytes = _ubuf.bytes();
+  int comm_bytes_per_rank = comm_bytes / _tp_size;
+
+  // TODO: Check if pull mode syncs before pulling the data
+  NVTE_CHECK(_ub_comm->push == 1,
+             "Userbuffer is populated on the send stream, so push mode must be enabled for proper "
+             "sequencing");
+  NVTE_CHECK(_tp_size == _ub_comm->nvsize,
```

Comment:
@timmoon10  Is there a good way to test these two cases to determine if I can remove these checks?

Reply:
Ok I updated it to use tp_size and tp_id from the communicator instead of relying on the comm->nvsize parameter